torch/csrc/jit/backends/backend_detail.cpp

```diff
@@ -124,6 +131,44 @@ Module codegen_backend_module(
             )",
       loweredModuleResolver());
 
+  // Debug info generated by BackendModuleDebugInfoRecorder
+  // WenNeed to have its lifetime tied to the module. It can either be in
+  // 1) Module or 2) In the TS module
+  // 1 requires us to introduce debug info that is backend specific into
+  // the Module class.
+  // So we opt for 2 at the moment. This is done as follows.
+  // We have a custom class BackendDebugInfo that stores debug info.
+  // This class is registered in lowered module and have a method
+  // that creates an instance of this class.
+  // We extract the instance and store debug info in it.
+  // This will be extracted at model saving time.
+  // This information will not be serialized and thus BackendDebugInfo.
+  // However a valid instance of BackendDebugInfo must exists when model is
+  // loaded. For example, say we embed a lowered module inside another python
+  // module. m = SomeModule(lowered_module) and then we do scripted_module =
+  // torch.jit.script(m) This step would fail, if valid instance of
+  // BackendDebugInfo existed in the model being scripted/serialized but did not
+  // exist when it was deserialized. Thus we create an instanace of
+  // BackendDebugInfo when backend is created in setstate. This instance will be
+  // just a dummy instance for lite interpreter (C10_MOBILE), which wont have
+  // same CPP class backing it. Unfortunately, these hacks are necessary in the
+  // absence of 1.
```

Comment:
@raziel by the way I added "For example, say we embed a lowered module inside another python module. m = SomeModule(lowered_module) and then we do scripted_module = torch.jit.script(m) This step would fail, if valid instance of BackendDebugInfo existed in the model being scripted/serialized but did not exist when it was deserialized. " because it the mentioned case did not fail then I did not have to introduce dummy type. I was able to get around it by just making debug_info type and optional type.

Reply:
I will remove that part because it seems that it may add more confusion than clarity.